### PR TITLE
Security flow

### PR DESCRIPTION
**Pull request by BMatijasevicExecom**
Related to https://github.com/execom-eu/exebite/issues/103#issuecomment-661743579

Implemented flow from related issue.
Added Identity Server Project.
Added OIDC Client js library to front-end and removed Angular social login library.
Added permissions and permission handler to Exebite API.
Refactored parts of Exebite API and front-end to conform to new security flows.

### CODE REVIEW

File: Exebite.IdentityServer/Config.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+                new Client {
+                    ClientId = "Exebite.ClientApp",
+                    ClientName = "Exebite Client App",
+                    AllowedGrantTypes = GrantTypes.Implicit,
+                    AllowedScopes = new List<string> { "openid" },
+                    RedirectUris = new List<string> { "http://localhost:4200/auth-callback" },
+                    PostLogoutRedirectUris = new List<string> { "http://localhost:4200" },
+                    AllowedCorsOrigins = new List<string> { "http://localhost:4200" },
+                    AllowAccessTokensViaBrowser = true,
+                }
```
Comment (delboj) [inline Exebite.IdentityServer/Config.cs:27]: Should we extract this configuration parameters into some kind of config file?

Comment (bdelicexecom) [inline Exebite.IdentityServer/Config.cs:27]: Should we extract this configuration parameters into some kind of config file? @BMatijasevicExecom

Comment (BMatijasevicExecom) [reply Exebite.IdentityServer/Config.cs:27]: I though so myself, but this is static class so I cannot use DI here... Should we change whole logic not to be static somehow?

Comment (bdelicexecom) [reply Exebite.IdentityServer/Config.cs:27]: If possible, it would be nice

Comment (BMatijasevicExecom) [reply Exebite.IdentityServer/Config.cs]: @bdelicexecom could you please revalidate?

**Pull request merged**